src/H5C.c

```diff
@@ -416,7 +416,7 @@ H5C_prep_for_file_close(H5F_t *f)
 
 #ifdef H5_HAVE_PARALLEL
     if ((H5F_INTENT(f) & H5F_ACC_RDWR) && !image_generated && cache_ptr->aux_ptr != NULL &&
-        f->shared->sblock && f->shared->fs_persist) {
```

Comment:
This just goes back to the previous behavior and tries to close as much of the cache as possible, even if the superblock isn't available.